packages/gatsby-cli/src/reporter/redux/internal-actions.ts

```diff
@@ -135,26 +152,18 @@ export const createPendingActivity = ({
   id: string
   status?: ActivityStatuses
 }): ActionsToEmit => {
-  const actionsToEmit: ActionsToEmit = []
-
-  const logsState = getStore().getState().logs
-
   const globalStatus = getGlobalStatus(id, status)
-
-  if (globalStatus !== logsState.status) {
-    actionsToEmit.push(setStatus(globalStatus))
-  }
-
-  actionsToEmit.push({
-    type: Actions.PendingActivity,
-    payload: {
-      id,
-      type: ActivityTypes.Pending,
-      status,
+  return [
+    setStatus(globalStatus),
+    {
+      type: Actions.PendingActivity,
+      payload: {
+        id,
+        type: ActivityTypes.Pending,
+        status,
+      },
     },
-  })
-
-  return actionsToEmit
+  ]
```

Comment:
gist of this and other activity related code changes is that we always would use `setStatus` action creator as it internally already checks for global status (or pending global status for delayed dispatchs)